core/src/main/scala/kafka/server/metadata/BrokerMetadataListener.scala

```diff
@@ -119,16 +119,24 @@ class BrokerMetadataListener(
       }
 
       _bytesSinceLastSnapshot = _bytesSinceLastSnapshot + results.numBytes
-      if (shouldSnapshot()) {
-        maybeStartSnapshot()
+      
+      val (takeSnapshot, snapshotReason) = shouldSnapshot()
+      if (takeSnapshot) {
+        maybeStartSnapshot(snapshotReason)
       }
 
       _publisher.foreach(publish)
     }
   }
 
-  private def shouldSnapshot(): Boolean = {
-    (_bytesSinceLastSnapshot >= maxBytesBetweenSnapshots) || metadataVersionChanged()
+  private def shouldSnapshot(): (Boolean, String) = {
```

Comment:
Here we can use an Option to represent the reason, similar to `QuorumFeatures.reasonNotSupported`.

Reply:
Thanks, I have made this change.